Handler/Demo.hs

```diff
@@ -0,0 +1,35 @@
+module Handler.Demo where
+
+import Import
+import Model.Site
+
+import Data.Maybe (fromJust, listToMaybe)
+
+getDemoR :: Handler Html
+getDemoR = do
+    root <- getAppRoot
+    siteId <- runDB $ findDemoSite root
+
+    defaultLayout $ do
+        setTitle "Carnival Demo"
+        $(widgetFile "demo")
+
+findDemoSite :: Text -> DB SiteId
+findDemoSite root = do
+    admin <- getAdmin
+    msite <- listToMaybe <$> findSites admin [SiteBaseUrl ==. root]
```

Comment:
I chose to re-use `findSites` vs extracting something more general that wouldn't be reused anyway. There is a `findOrInsert` in persistent already I think, but we couldn't use it because of the two tables involved (`site`, and `membership`). I can't picture a `findOrInsert` of much generic value that correctly handles the "join" lookup.